kratos/python_scripts/analysis_stage.py

```diff
@@ -211,6 +211,20 @@ def ChangeMaterialProperties(self):
         """this function is where the user could change material parameters as a part of the solution step """
         pass
 
+    def Save(self, serializer : KratosMultiphysics.StreamSerializer):

```

Comment:
about the first point i tend to disagree. We need to serialize this to a list of characters to later pickle it. for this the StreamSerializer is the proper object...

Reply:
about the first point i tend to disagree. We need to serialize this to a list of characters to later pickle it. for this the StreamSerializer is the proper object...

Reply:
- About 1: see @RiccardoRossi 's comment.
- About 2: Basically I forgot to add the return types. I'll do it right away.

Reply:
@RiccardoRossi I don't get your point. Are you using some non-base class interface?

Reply:
the point is not that we cannot use the base class serializer. It would work.

The point is that we specifically want to pass a StreamSerializer to later pickle it, so the current code is doing exactly what we expect.